tls_main.c

```diff
@@ -420,6 +426,18 @@ static int do_tls_getsockopt_tx(struct sock *sk, char __user *optval,
 	return rc;
 }
 
+static int do_tls_getsockopt_lua(struct sock *sk, char __user *optval,
+				 int __user *optlen)
+{
+	struct tls_context *ctx = tls_get_ctx(sk);
+
+	if (optlen != sizeof(int))
+		return -EINVAL;
+
+	copy_to_user(optval, &ctx->lua_err, sizeof(int));
```

Comment:
I don't know if we should do this now, but it would be interesting if we could also send to user space the Lua recv function's returned values. @tammela what do you think?

Reply:
In the future, sure. For now let's stick with this simple mechanism.